sdk/client_test.go

```diff
@@ -0,0 +1,67 @@
+package sdk
+
+import (
+	"net/http"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+const (
+	authHeader = "Authorization"
+	authPrefix = "Bearer "
+)
+
+func TestHeaders_Add(t *testing.T) {
+	t.Parallel()
+
+	Convey("given the sdk Headers struct contains a value for ETag", t, func() {
+		req := &http.Request{}
+		headers := &Headers{
+			ETag: "dsalfhjsadf",
+		}
+
+		Convey("when calling the Add method on Headers", func() {
+			headers.Add(req)
+
+			Convey("then an ETag header is set on the request", func() {
+				So(req.Header, ShouldBeEmpty)
+			})
+		})
+	})
+
+	Convey("given the sdk Headers struct contains avalue for If-Match", t, func() {
+		req := &http.Request{}
+		headers := &Headers{
+			IfMatch: "*",
+		}
+
+		Convey("when calling the Add method on Headers", func() {
+			headers.Add(req)
+
+			Convey("then an If-Match header is set on the request", func() {
+				So(req.Header, ShouldBeEmpty)
```

Comment:
Similarly I think it would be good to add a TODO here. Otherwise this test is passing, even though it should be failing as adding the If-Match header has not been implemented yet, and there's nothing explaining why.

Reply:
nope, same as above, the test will fail once implemented

Reply:
I realise that the test will fail once implemented. Similarly to above, I was trying to suggest that you add a TODO comment to explain that this will need to be updated once the If-Match is implemented. But I don't mind if you leave it as is.